### PR TITLE
Ana/fix publish flow

### PR DESCRIPTION
**Pull request by Bitcoinera**
Closes #ISSUE

**Description:**

Let's see

Thank you! 🚀

---

For contributor:

- [ ] Added changes entries. Run `yarn changelog` for a guided process.
- [ ] Reviewed `Files changed` in the github PR explorer
- [ ] Attach screenshots of the UI components on the PR description (if applicable)
- [ ] Scope of work approved for big PRs

For reviewer:

- [ ] Manually tested the changes on the UI

### CODE REVIEW

**Comment by Bitcoinera** [comment]
Oops, it didn't trigger "Publish to master"... why

**Comment by faboweb** [comment]
🚀 Deployed on https://5ef4db7f4f87693ddf64360e--lunieio.netlify.app

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -22,4 +17,6 @@ jobs:
         - name: Use Node.js
           uses: actions/setup-node@v1
         - name: publish to master
-          run: bash ./scripts/publish.sh        
\ No newline at end of file
+          run: bash ./scripts/publish.sh $GIT_BOT_TOKEN
```
Comment (faboweb) [inline .github/workflows/publish.yml:20]: nice. we could also just get content of publish.sh and put it in here. what do you think?

Comment (Bitcoinera) [reply .github/workflows/publish.yml:20]: yeah, that would work too. Maybe better and then have it all in one single place?

**Comment by Bitcoinera** [comment]
Should we rebase to handle conflicts? I will investigate:

```
CONFLICT (content): Merge conflict in extension/src/popup/gql/apollo.js
Removing babel.config.js
Removing android/capacitor.settings.gradle
Removing .github/pr_alert.yml
Automatic merge failed; fix conflicts and then commit the result.
##[error]Process completed with exit code 1.
```

**Comment by faboweb** [comment]
There should not a be conflicts when merging to master as the flow is linear. If there are conflicts we need to investigate manually. Probably need to merge master into develop before.

**Comment by Bitcoinera** [comment]
> There should not a be conflicts when merging to master as the flow is linear. If there are conflicts we need to investigate manually. Probably need to merge master into develop before.

No, I realized this was because I was using a very strange branch from January 😅  Don't worry

**Comment by Bitcoinera** [comment]
Should I add changelogs for `app` and `extension` too? It feels kind of repetitive... 😅

File: extension/scripts/publish.sh [hunk 2]
```diff
@@ -3,6 +3,7 @@ COMMIT_MESSAGE=$(git log --format=%B -n 1)
 if [[ $COMMIT_MESSAGE == "release-"* || $COMMIT_MESSAGE == "[Simsala] automatic release"* ]]
 then
     echo "Publishing"
+    git config pull.rebase false
```
Comment (faboweb) [inline extension/scripts/publish.sh:6]:
```suggestion
```

I think that is the default

File: .github/workflows/publish.yml [hunk 3]
```diff
@@ -22,4 +22,20 @@ jobs:
...
+            then
+              echo "Publishing" &&
+              git config pull.rebase false &&
+              git config user.email "bot@lunie.io" &&
+              git config user.name "Publish Lunie Bot" &&
+              git remote add bot https://${GIT_BOT_TOKEN}@github.com/luniehq/lunie.git &&
+              git checkout master &&
+              git pull &&
+              git merge --no-edit origin/develop &&
+              git push;
```
Comment (faboweb) [inline .github/workflows/publish.yml:36]: do we need the `&&`?

File: extension/scripts/publish.sh (hunk 2 above)
Comment (Bitcoinera) [reply extension/scripts/publish.sh:6]: yeah, but we get the warning logs, which are quite annoying and recommend doing this. Better leave this here I think

File: .github/workflows/publish.yml (hunk 3 above)
Comment (Bitcoinera) [reply .github/workflows/publish.yml:36]: yes, I tried it. It is because here it is not a script but directly command line

**Pull request merged**